### ISSUE TITLE
Screenshots of webpages aren't as accurate as they could be

### ISSUE THREAD

**Issue by pamelafox** [Khan/live-editor#420]
In the ProcessingJS environment, we can capture a screenshot that looks exactly like the canvas. In the webpage environment, we use a library that makes an attempt at capturing what the webpage looks like, but fails in many ways.

Darryl reported a few via ZenDesk:
"Screenshots of HTML programs are not reflective of its CSS styling, often positioning elements in strange places, and ignoring any text set in a Google Font, externally loaded image, or background color assigned to the <body> tag.

There are plenty of examples to show this, but here are a few:

Program: 
https://www.khanacademy.org/computer-programming/scrollable-star-wars-crawl/4900272747970560 
Screenshot: 
https://www.khanacademy.org/computer-programming/scrollable-star-wars-crawl/4900272747970560/latest.png

Program: 
https://www.khanacademy.org/computer-programming/all-my-programs-filterable/6171340237307904 
Screenshot: 
https://www.khanacademy.org/computer-programming/all-my-programs-filterable/6171340237307904/latest.png

Program: 
https://www.khanacademy.org/computer-programming/new-personal-website/5468974101561344 
Screenshot: 
https://www.khanacademy.org/computer-programming/new-personal-website/5468974101561344/latest.png

Program: 
https://www.khanacademy.org/computer-programming/transcolors-webpage/6049691310751744 
Screenshot: 
https://www.khanacademy.org/computer-programming/transcolors-webpage/6049691310751744/latest.png"

If anyone wants to take a stab at improving even one of those aspects of webpage screenshots, PRs are welcome. :)

**Comment by jaredly** [comment]
Are the screenshots done using [html2canvas](http://html2canvas.hertzen.com/) or similar? Unless you're doing offline screenshotting using phantomjs or similar, it's really hard to get perfect fidelity :( I've looked into this before...
The biggest limitation of client-side javascript screenshotting is that you can't use external resources at all (e.g. Google Fonts or external images).

**Comment by pamelafox** [comment]
Yeah, I believe that's it. The images are all on kasandbox.org or wikipedia.org, so one potential thing is to proxy them via khanacademy.org (or look to see if we can change a setting in server headers that make <canvas> okay with accessing them-- I'm not super up to date on the canvas/img permissions).

**Comment by jaredly** [comment]
Looks like CORS-enabled images don't taint the canvas
https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image

**Comment by pamelafox** [comment]
Sweet, that's what I was hoping. I think that'd be a change on our varnish
server config (which proxies S3 files), so that particular change can't be
done in this repo.
